code/game/objects/structures/window.dm

```diff
@@ -419,7 +419,6 @@
 
 	var/ratio = atom_integrity / max_integrity
 	ratio = CEILING(ratio*4, 1) * 25
-	cut_overlay(crack_overlay)
```

Comment:
Something isn't right. We need this to remove the old overlay and add the new one when the damage ratio changes else they will just stack on each other giving a very cluttered appearance.

This one line has been around for years and has worked and the fact that now it is causing problems is revealing a deeper problem.

@LemonInTheDark thoughts?

Reply:
This is the function which updates overlays now.
https://github.com/tgstation/tgstation/blob/433718c545be518b6b5bb52369dfed6a2bc76d7f/code/game/atom/atom_appearance.dm#L36-L99

Since https://github.com/tgstation/tgstation/pull/78719 overlay update is only called if the managed overlays are updated, i.e. the overlay we're calling for next isn't the same as the overlay we're currently managing. In the case where we're asking it to update the same overlay again and again it won't contiously make more overlays. When the damage ratio changes we ask for a new overlay and a new mutable appearance and so it updates but only once. The screenshot and after video seem to suggest this behavior since I var edit'ed a window to show that there's only ever 2 overlays at a time even between different damage ratios. Though there is a very real chance I am ending up misinterpreting the code to fit my soloution, since I am somewhat still new to this. I appreciate the quick response anyways.

Reply:
This looks like legacy code. The old pattern was to add overlays individually, and remove them individually. This is error prone tho, and leads to messy code.
The current pattern is to, inside update_overlays(), create a mutable appearance, and add it to . (a list here)
atoms keep track of overlays added via this method, so if an overlay ISN'T in the list, it's removed, and otherwise it's added.

what you're seeing is likely boomer code. Best practice would be to not keep crack_overlay as a var on the window at all, and keep the mutable_appearance local to the update_overlays override.

Reply:
I am unsure of the error you were seeing though. do you have an example of it?